src/fsharp/fsi/fsi.fs

```diff
@@ -613,13 +613,23 @@ type internal FsiCommandLineOptions(fsi: FsiEvaluationSessionHostConfig, argv: s
     let isInteractiveServer() = fsiServerName <> ""  
     let recordExplicitArg arg = explicitArgs <- explicitArgs @ [arg]
 
-    let executableFileName = 
-        lazy 
-            match tcConfigB.exename with
-            | Some s -> s
-            | None -> 
-            let currentProcess = System.Diagnostics.Process.GetCurrentProcess()
-            Path.GetFileName(currentProcess.MainModule.FileName)
+    let executableFileNameWithoutExtension =
+        lazy
+            let getFsiCommandLine () =
+                let fileNameWithoutExtension path = Path.GetFileNameWithoutExtension(path)
```

Comment:
fsi is not named fsi only linux, it is fsharpi.  Looking at the original code, it was intended to capture that.

I think the check should simply be:  is the current process dotnet.exe, if so then use dotnet fsi, otherwise use the process name.

At some time in the near future, we are going to go back to fsi.exe starting the coreclr version of fsi too :-)

Reply:
Unfortunately this also changed the output on the desktop version and fails the test:

Originally:
````
Usage: fsi.exe <options> [script.fsx [<arguments>]]
````
With this change
````
Usage: fsi <options> [script.fsx [<arguments>]]
````

https://github.com/dotnet/fsharp/blob/master/tests/fsharpqa/Source/CompilerOptions/fsi/help/help40-nologo.437.1033.bsl

https://github.com/dotnet/fsharp/blob/master/tests/fsharpqa/Source/CompilerOptions/fsi/help/help40.437.1033.bsl

If you want, I can take care of fixing this when I do fsi,exe starting the coreclr version.

Reply:
> I think the check should simply be: is the current process dotnet.exe, if so then use dotnet fsi, otherwise use the process name.

@KevinRansom:
That's [what I started doing](https://github.com/dotnet/fsharp/pull/9066#issuecomment-620863919), but then I thought that maybe in the future it might be renamed to `dnet` (or whatever) and I wanted to be safe from such a change. What do you think?

I aligned the two outputs from `dotnet fsi --help` and `dotnet fsi; #help;;`, does it make sense to change the test in this case? I don't see a benefit of having `.exe` in the `--help` version. Let me know what you think.